core/src/main/java/org/infinispan/security/impl/AuthorizationHelper.java

```diff
@@ -61,11 +59,16 @@ public void checkPermission(AuthorizationPermission perm, String role) {
    }
 
    public void checkPermission(AuthorizationConfiguration configuration, AuthorizationPermission perm) {
-      checkPermission(configuration, perm, null);
+      checkPermission(configuration, perm, (String)null);
    }
 
    public void checkPermission(AuthorizationConfiguration configuration, AuthorizationPermission perm,
-         String role) {
+                               String role) {
+      checkPermission(configuration, perm, Collections.singletonList(role));
```

Comment:
what would you say for changing this method to:

public void checkPermission(AuthorizationConfiguration configuration, AuthorizationPermission perm,
                               String role) {
      List<String> roles = role == null ? Collections.emptyList() : Collections.singletonList(role);
      checkPermission(configuration, perm, roles);
   }